core/thread/inc/ROOT/TReentrantRWLock.hxx

```diff
@@ -144,8 +144,93 @@ struct RecurseCounts {
    void ResetIsWriter(local_t & /* local */) { fWriterThread = std::thread::id(); }
 
    size_t &GetLocalReadersCount(local_t &local) { return fReadersCount[local]; }
+};
+
+struct RecurseCountsShared {
```

Comment:
When adding a new count options, we should extent the test ` core/thread/test/testRWLock.cxx `
Starting with adding a new global variable to the section:
```
auto gMutex = new TMutex(kTRUE);
auto gRWMutex = new TRWMutexImp<TMutex>();
auto gRWMutexSpin = new TRWMutexImp<ROOT::TSpinMutex>();
auto gRWMutexStd = new TRWMutexImp<std::mutex>();
auto gReentrantRWMutex = new ROOT::TReentrantRWLock<TMutex>();
auto gReentrantRWMutexSM = new ROOT::TReentrantRWLock<ROOT::TSpinMutex>();
auto gReentrantRWMutexStd = new ROOT::TReentrantRWLock<std::mutex>();
auto gSpinMutex = new ROOT::TSpinMutex();

// Intentionally ignore the Fatal error due to the shread thread-local storage.
// In this test we need to be 'careful' to not use all those mutex at the same time.
int trigger1 = gErrorIgnoreLevel = kFatal + 1;
auto gReentrantRWMutexTL = new ROOT::TReentrantRWLock<TMutex, ROOT::Internal::UniqueLockRecurseCount>();
auto gReentrantRWMutexSMTL = new ROOT::TReentrantRWLock<ROOT::TSpinMutex, ROOT::Internal::UniqueLockRecurseCount>();
auto gRWMutexTL = new TRWMutexImp<TMutex, ROOT::Internal::UniqueLockRecurseCount>();
auto gRWMutexTLSpin = new TRWMutexImp<ROOT::TSpinMutex, ROOT::Internal::UniqueLockRecurseCount>();
int trigger2 = gErrorIgnoreLevel = 0;

```
(in the right section, like the one between trigger1 and trigger2 in this case).

Beside testing the functionality it also give a first level benchmark of the raw speed of the options.

Reply:
Added